rj_utils/include/rj_utils/concurrent_queue.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <condition_variable>
+#include <mutex>
+#include <queue>
+
+namespace rj_utils {
+/**
+ * @brief A SPSC blocking queue that is unbounded in size, using std::queue
+ * as the internal backing data structure.
+ * @tparam T The type to be used in the queue.
+ */
+template <typename T>
+class ConcurrentQueue {
+public:
+    /**
+     * @brief Blocking get.
+     * @return The first item in the queue, blocking until one is available.
+     */
+    T Get() {
+        std::unique_lock<std::mutex> lock{mutex_};
+        cv_.wait(lock, [this]() { return !queue_.empty(); });
+        auto item = std::move(queue_.front());
+        queue_.pop();
+        return item;
+    }
+
+    bool TryGet(T& item, const std::chrono::milliseconds& timeout) {
```

Comment:
Really this should just return an `std::optional`, right? Whatever, fine for now.